lib/cuda/cudaDeviceInterface.hpp

```diff
@@ -53,6 +53,13 @@ class cudaDeviceInterface final : public gpuDeviceInterface {
     void async_copy_host_to_gpu(void* dst, void* src, size_t len, cudaEvent_t pre_event,
                                 cudaEvent_t& copy_pre_event, cudaEvent_t& copy_post_event);
 
+    /**
+    * @brief Print avialable colcks for given GPU and set sevice clock
+    * @param unique_name Base path from CudaProcess
+    * @param gpu_id_  Device ID
+    */
+    void cudaDeviceInterface::set_device_clocks(const string& unique_name, int32_t gpu_id_);
```

Comment:
Since this is member function you can use the `gpu_id` variable from the base class directly and don't need to pass it into this function. 

Similarly I think we should make `unique_name` a member variable of this class, and then we won't need to pass that around either.

Reply:
Done